fbw-common/src/wasm/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -242,17 +245,55 @@ impl<const ZONES: usize, const ENGINES: usize> AirConditioningSystemController<Z
     pub fn individual_pack_flow(&self) -> MassRate {
         self.pack_flow_controller.pack_flow()
     }
+
+    pub fn trim_air_pressure_regulating_valve_controller(
+        &self,
+    ) -> TrimAirPressureRegulatingValveController {
+        self.trim_air_system_controller.taprv_controller()
+    }
+
+    pub fn trim_air_pressure_regulating_valve_is_open(&self) -> bool {
+        self.trim_air_system_controller.tarpv_is_open()
+    }
+
+    pub fn duct_overheat(&self, zone_id: usize) -> bool {
+        self.trim_air_system_controller.duct_overheat(zone_id)
+    }
+
+    pub fn hot_air_pb_fault_light_determination(&self) -> bool {
+        self.trim_air_system_controller.duct_overheat_monitor()
+    }
+
+    pub fn galley_fan_fault(&self) -> bool {
+        self.zone_controller
+            .iter()
+            .any(|zone| zone.galley_fan_fault())
+    }
+
+    pub fn taprv_position_disagrees(&self) -> bool {
+        self.trim_air_system_controller
+            .taprv_disagree_status_monitor()
+    }
 }
 
 impl<const ZONES: usize, const ENGINES: usize> DuctTemperature
     for AirConditioningSystemController<ZONES, ENGINES>
 {
     fn duct_demand_temperature(&self) -> Vec<ThermodynamicTemperature> {
-        let mut duct_demand_temperature = Vec::new();
-        for controller in self.zone_controller.iter() {
-            duct_demand_temperature.extend(&controller.duct_demand_temperature())
-        }
-        duct_demand_temperature
+        let demand_temperature: Vec<ThermodynamicTemperature> = self
+            .zone_controller
+            .iter()
+            .map(|zone| zone.duct_demand_temperature()[0])
+            .collect();
+
+        // Because each ACSC calculates the demand of its respective zone(s), we fill the vector for the trim air system
+        let mut filler_vector = [ThermodynamicTemperature::new::<degree_celsius>(24.); ZONES];
+        if matches!(self.id, AcscId::Acsc1(_)) {
+            filler_vector[..1].copy_from_slice(&demand_temperature);
+        } else {
+            filler_vector[1..].copy_from_slice(&demand_temperature);
+        };
+        filler_vector.to_vec()
```

Comment:
```suggestion
        let demand_temperature = self
            .zone_controller
            .iter()
            .map(|zone| zone.duct_demand_temperature()[0]);

        // Because each ACSC calculates the demand of its respective zone(s), we fill the vector for the trim air system
        if matches!(self.id, AcscId::Acsc1(_)) {
            demand_temperature.take(ZONES).collect()
        } else {
            demand_temperature.skip(1).take(ZONES).collect()
        }
```

Reply:
This suggestion doesn't work. The problem is that the resulting vector here needs to be the length of `ZONES` but `zone_controller` is a different length in each ACSC. For the a320:
- There are 3 ZONES
- ACSC1 has a `zone_controller` of length 1 (cockpit)
- ACSC2 has a `zone_controller` of length 2 (cabin zones)

What we want is to "fill" the remaining positions of the resulting vector with dummy data, so for ACSC1 you end up with [real, dummy dummy] and for ACSC2 [dummy, real, real]. Hope this makes sense, but I'm open to other suggestions as I don't particularly like this piece of code either!

Reply:
Does it have to return ZONES entries? Because actually the function return type doesn't specify that. (I know it was a constant size array before, but now the caller of this function cannot rely on that)

Reply:
Yeah I see your point... I could introduce const generics in the trait itself and then return an array `[ThermodynamicTemperature; ZONES]` instead of a vector. But this makes the whole code a lot more verbose because I need to pass the generics around more elements and/or hard code the number of zones, which I don't really like either. I could add a comment explaining why this function needs to return a vector with the length of ZONES here. Or do you think it's worth it having the generics in the return type of this trait?

Reply:
Comment added 😃. Thanks again @Gurgel100